pkg/test/framework/components/ingress/kube.go

```diff
@@ -42,65 +45,102 @@ var (
 )
 
 type kubeComponent struct {
-	id      resource.ID
-	address string
+	id          resource.ID
+	address     string
+	gatewayType IgType
+	caCert      string
+	tlsCert     string
+	tlsKey      string
 }
 
-func newKube(ctx resource.Context, cfg Config) (Instance, error) {
-	c := &kubeComponent{}
-	c.id = ctx.TrackResource(c)
+// getHTTPAddress returns the ingress gateway address for plain text http requests.
+func getHTTPAddress(env *kube.Environment, cfg Config) (interface{}, bool, error) {
+	// In Minikube, we don't have the ingress gateway. Instead we do a little bit of trickery to to get the Node
+	// port.
+	n := cfg.Istio.Settings().IngressNamespace
+	if env.Settings().Minikube {
+		pods, err := env.GetPods(n, fmt.Sprintf("istio=%s", istioLabel))
+		if err != nil {
+			return nil, false, err
+		}
 
-	env := ctx.Environment().(*kube.Environment)
+		scopes.Framework.Debugf("Querying ingress, pods:\n%v\n", pods)
+		if len(pods) == 0 {
+			return nil, false, errors.New("no ingress pod found")
```

Comment:
let's be consistent of using `errors.New` or `fmt.Errorf`

Reply:
There are four fmt.Errorf and four errors.New. Changed errors.New to fmt.Errorf.